docs/madr/decisions/053-sni.md

```diff
@@ -31,23 +31,22 @@ It's important to note that we never parse SNI. We only use it for multizone com
 It means that we can hash the data as long as it's consistent hashing.
 We cannot put tags in the original form because they can violate hostname requirements. For example `/` in `kuma.io/service` is not ok.
 The new format could look like this then:
-`<resource-name>.<mesh-name>.<extra-hash-if-needed>.<resource-type>`
-* `<resource-name>` - (required) name of the resource. Name of MeshService or MeshExternalService
-* `<mesh-name>` - (required) name of the mesh
-* `<extra-hash-if-needed>` - (optional) 16 characters of SHA256. In case of MeshService tags it's `SHA256(<tag1>=<value1>,<tag2>=<value2>)` sorted by keys.
-* `<resource-type>` - (required) `ms` (MeshService) or `mes` (MeshExternalService)
+`<format-version><hash>.<resource-name>.<mesh-name>.<resource-type>`
+* `<format-version>` - (required) `a`. 1 character. It's so we can evolve format in the future.
+* `<hash>` - (required) 16 characters of fnv64. It's `FNV64(resourceName,mesh,type,optional-data)`. Optional data can be for example tags.
+* `<resource-name>` - (required) name of the resource. Name of MeshService or MeshExternalService.
+  Trimmed to 128 chars. If trimmed we add `x` at the end, so we don't end up in invalid hostname `verylongname-.default.ms` 
+* `<mesh-name>` - (required) name of the mesh.
+  Trimmed to 64 chars. If trimmed we add `x` at the end.
+* `<resource-type>` - (required) `ms` (MeshService) or `mes` (MeshExternalService). Abbreviation to at most 3 chars. 
 
-Examples:
-* `backend.demo.ms` - MeshService named `backend` in `demo` mesh
-* `redis.kuma-demo.default.ms` - MeshService named `redis` in `kuma-demo` namespace in `default` mesh
-* `google.default.mes` - MeshExternalService named `google` in `default` mesh
-* `backend.demo.e3b0c44298fc1c.ms` - subset of MeshService named `backend` in `demo` mesh for tags `version=v2`
+The proposed format has at most 1+16+129+65+3 = 214 chars.
```

Comment:
129 and 65 here because of the `x` at the end? if so, then we also need to count dots no?

Reply:
yes, updated